packages/shared/docs/README.md

```diff
@@ -0,0 +1 @@
+### Architecture and general philosophy of this package
```

Comment:
WIP?

Reply:
Yes, but not in this PR. We need more documentation content for these files I have created, so feel free to add content after this PR is merged.